org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/WorkspaceEncodingProvider.java

```diff
@@ -47,30 +48,27 @@
 
 	@Override
 	public String getEncoding(URI uri) {
-		if (workspace != null) {
-			if (uri != null) {
-				Iterator<Pair<IStorage, IProject>> storages = storage2UriMapper.getStorages(uri).iterator();
-				while (storages.hasNext()) {
-					Pair<IStorage, IProject> storage = storages.next();
-					if (storage.getFirst() instanceof IEncodedStorage) {
-						try {
-							return ((IEncodedStorage) storage.getFirst()).getCharset();
-						} catch (CoreException e) {
-							LOG.error("Error getting file encoding", e);
-						}
-					}
+		if (workspace != null && uri != null) {
```

Comment:
The only semantic difference seems to be, that on uri==null it will no longer return the default charset of the workspace. Why is that desirable?